src/storage/resourceManager.ts

```diff
@@ -198,11 +208,12 @@ export class ResourceManager {
 
   // ENVIRONMENTS (coarse resource)
 
-  async setEnvironments<T extends EnvironmentType>(
-    connectionId: ConnectionId,
-    environments: T[],
-  ): Promise<void> {
-    await this.storeCoarseResources(connectionId, CoarseResourceKind.ENVIRONMENTS, environments);
+  async setEnvironments(connectionId: ConnectionId, environments: Environment[]): Promise<void> {
```

Comment:
The store paths don't need to be generic, in that they never actually used T, so base class Environment is fine. We only need to use T when loading back from storage.